app/lib/repository.rb

```diff
@@ -456,8 +456,12 @@ def add(path, file_data=nil, mime_type=nil)
       @jobs.push(action: :add, path: path, file_data: file_data, mime_type: mime_type)
     end
 
-    def remove(path, expected_revision_identifier)
-      @jobs.push(action: :remove, path: path, expected_revision_identifier: expected_revision_identifier)
+    def remove(path, expected_revision_identifier, option)
+      @jobs.push(action: :remove, path: path, expected_revision_identifier: expected_revision_identifier, option: option)
+    end
+
+    def remove_directory(path, expected_revision_identifier)
```

Comment:
If `GitRepository` is the only class that will really use `remove_repository`, could we instead make `GitRepository.commit` a bit smarter so that we don't need `remove_repository` and we could just make `remove` work for both files and folders (like it does for `MemoryRepository` and `SubversionRepository`)?

Reply:
Nevermind